src/cartesian/Line.tsx

```diff
@@ -184,6 +184,17 @@ export class Line extends PureComponent<Props, State> {
     this.setState({ totalLength });
   }
 
+  componentDidUpdate(): void {
+    if (!this.props.isAnimationActive) {
+      return;
+    }
+
+    const totalLength = this.getTotalLength();
+    if (totalLength != null && totalLength !== this.state.totalLength) {
```

Comment:
This part feels redundant. Since the null case does not exist and it's a primitive type, it seems fine to use `this.setState({ totalLength: this.getTotalLength() });`

Reply:
The tooltip inherits PureComponent, it already performs shallow comparisons.
Primitive type references always remain the same, so it won't rerender, but not calling it explicitly also seems good